types/react/test/index.ts

```diff
@@ -743,3 +757,36 @@ class RenderChildren extends React.Component {
         return children !== undefined ? children : null;
     }
 }
+
+{
+// React 16 - class components created via Component/PureComponent are required to implement render method and return ReactNode
+// https://reactjs.org/docs/react-component.html#render
+
+// $ExpectError
+class MissingRenderer extends React.Component {}
+class MissingRenderer2 extends React.Component {
+    // $ExpectError
```

Comment:
I needed to bump TypeScript version to 2.7 because this test was failing with TS 2.6 no idea why, any suggestions?

With that [alt](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/alt/index.d.ts) is now failing on CI as it depends on react and TS >= 2.6 .

Any suggestions how to resolve this are more than welcome ! 

< maybe bumping alt to TS 2.7 as well ?

Reply:
What's the error that is given when you are using TS <= 2.6?

Reply:
![image](https://user-images.githubusercontent.com/1223799/41684191-73cfeae2-74e5-11e8-8479-cf2501e88cac.png)